### ISSUE TITLE
Error creating SSM document: InvalidDocumentSchemaVersion: Document type is required for schema version 2.0

### ISSUE THREAD

**Issue by DenisBY** [hashicorp/terraform#11788]
### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

```
$  terraform -v
Terraform v0.8.6
```

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_ssm_document

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
provider "aws" {
    region = "${var.region}"
}

data "template_file" "document" {
    template = "${file("${path.module}/amiwindowsupdate.json")}"
}

resource "aws_ssm_document" "document" {
  name    = "WindowsUpdate",
  content = "${data.template_file.document.rendered}"
}
```

```json
{
   "schemaVersion":"2.0",
   "description":"Run a script",
   "parameters":{
      "commands":{
         "type":"StringList",
         "description":"(Required) Specify a shell script or a command to run.",
         "minItems":1,
         "displayType":"textarea"
      }
   },
   "mainSteps":[
      {
         "action":"aws:runShellScript",
         "name":"runShellScript",
         "inputs":{
            "commands":"{{ commands }}"
         }
      },
      {
         "action":"aws:runPowerShellScript",
         "name":"runPowerShellScript",
         "inputs":{
            "commands":"{{ commands }}"
         }
      }
   ]
}
```

### Expected Behavior
What should have happened?

Document created

### Actual Behavior
What actually happened?

```
Error applying plan:

1 error(s) occurred:

* aws_ssm_document.document: [ERROR] Error creating SSM document: InvalidDocumentSchemaVersion: Document type is required for schema version 2.0
        status code: 400, request id: 261ef6f7-ee1b-11e6-b12a-27e73feaa5c3
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

I think AWS added `document-type` as mandatory option not so long time ago as even its documentation doesn't have this option:
http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/create-ssm-doc.html#create-ssm-console-cli

**Comment by liamjbennett** [comment]
@DenisBY this one's on me. Will be a PR to fix soon.

**Comment by DenisBY** [comment]
I guess it's fixed by this PR?
https://github.com/hashicorp/terraform/pull/11671

**Comment by stack72** [comment]
Hi @DenisBY 

Not sure if you are in a position to create a new terraform binary and test this for us - but i believe this issue is fixed in master by #11671 

Thoughts?

Paul

**Comment by DenisBY** [comment]
Yes, I was able to use SSM with version 0.8.7, so it's fixed.

**Comment by stack72** [comment]
Closed via #11671
